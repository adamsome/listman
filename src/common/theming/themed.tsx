import React from 'react'
import { THEME_WHITE } from '.'
import ThemeProvider from './ThemeProvider'

const themed = (children: React.ReactNode) => {
  return <ThemeProvider theme={THEME_WHITE}>{children}</ThemeProvider>
}

export default themed
