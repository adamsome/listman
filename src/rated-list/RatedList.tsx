import React from 'react'
import Info from '../common/Info'
import ArtifactRow from './ArtifactRow'
import RatingRow from './RatingRow'
import { RatedListItem } from './types'

type Props = typeof defaultProps & {
  items: RatedListItem[]
}

const defaultProps = {}

const RatedList = (props: Props): JSX.Element => {
  const { items } = props
  return <div>{items.length ? items.map(rowTypeChooser) : none}</div>
}

const rowTypeChooser = (item: RatedListItem) => {
  switch (item.type) {
    case 'artifact':
      return <ArtifactRow key={item.id} item={item} />
    case 'rating':
      const key = item.rating != null ? item.rating : -1
      return <RatingRow key={key} item={item} />
  }
}

const none = <Info>No artifacts</Info>

RatedList.defaultProps = defaultProps

export default RatedList
