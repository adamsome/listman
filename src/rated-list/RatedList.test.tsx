import React from 'react'
import ReactDOM from 'react-dom'
import themed from '../theming/themed'
import RatedList from './RatedList'
import { SHORT_LIST } from './RatedList.stories'

export const actions = {
  onMove: () => null,
}

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(themed(<RatedList items={SHORT_LIST} {...actions} />), div)
  ReactDOM.unmountComponentAtNode(div)
})
