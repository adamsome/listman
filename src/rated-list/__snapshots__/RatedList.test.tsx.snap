// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.emotion-1 {
  color: hsl(0,0%,13%);
  background-color: hsl(0,0%,100%);
}

.emotion-0 {
  padding-left: 5rem;
  margin-top: 2rem;
  margin-bottom: 1.25rem;
}

<div
  className="emotion-1"
>
  <div>
    <div
      className="emotion-0"
    >
      <div>
        <svg
          height="3rem"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "cursor": "auto",
            }
          }
          width="3rem"
        >
          <g
            style={
              Object {
                "opacity": 0,
                "transition": "opacity 100ms ease-in-out",
              }
            }
            x={-0.2}
            y={0.9}
          >
            <svg
              viewBox="0 0 512 512"
            >
              <circle
                cx="256"
                cy="256"
                fill="currentColor"
                r="40"
                strokeWidth="0"
                transform="inherit"
                transform-origin="center"
              />
            </svg>
          </g>
          <g
            style={
              Object {
                "opacity": 1,
                "transform": "scale(1)",
                "transformOrigin": "center",
                "transition": "opacity 200ms ease-in-out, transform 100ms ease-in-out",
              }
            }
          >
            <svg
              viewBox="0 0 51 48"
            >
              <path
                d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
                fill="currentColor"
                stroke="currentColor"
              />
            </svg>
          </g>
        </svg>
        <svg
          height="3rem"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "cursor": "auto",
            }
          }
          width="3rem"
        >
          <g
            style={
              Object {
                "opacity": 0,
                "transition": "opacity 100ms ease-in-out",
              }
            }
            x={-0.2}
            y={0.9}
          >
            <svg
              viewBox="0 0 512 512"
            >
              <circle
                cx="256"
                cy="256"
                fill="currentColor"
                r="40"
                strokeWidth="0"
                transform="inherit"
                transform-origin="center"
              />
            </svg>
          </g>
          <g
            style={
              Object {
                "opacity": 1,
                "transform": "scale(1)",
                "transformOrigin": "center",
                "transition": "opacity 200ms ease-in-out, transform 100ms ease-in-out",
              }
            }
          >
            <svg
              viewBox="0 0 51 48"
            >
              <path
                d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
                fill="currentColor"
                stroke="currentColor"
              />
            </svg>
          </g>
        </svg>
        <svg
          height="3rem"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "cursor": "auto",
            }
          }
          width="3rem"
        >
          <g
            style={
              Object {
                "opacity": 0,
                "transition": "opacity 100ms ease-in-out",
              }
            }
            x={-0.2}
            y={0.9}
          >
            <svg
              viewBox="0 0 512 512"
            >
              <circle
                cx="256"
                cy="256"
                fill="currentColor"
                r="40"
                strokeWidth="0"
                transform="inherit"
                transform-origin="center"
              />
            </svg>
          </g>
          <g
            style={
              Object {
                "opacity": 0.25,
                "transform": "scale(1)",
                "transformOrigin": "center",
                "transition": "opacity 200ms ease-in-out, transform 100ms ease-in-out",
              }
            }
          >
            <svg
              viewBox="0 0 51 48"
            >
              <path
                d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
                fill="currentColor"
                stroke="currentColor"
              />
            </svg>
          </g>
        </svg>
        <svg
          height="3rem"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "cursor": "auto",
            }
          }
          width="3rem"
        >
          <g
            style={
              Object {
                "opacity": 0,
                "transition": "opacity 100ms ease-in-out",
              }
            }
            x={-0.2}
            y={0.9}
          >
            <svg
              viewBox="0 0 512 512"
            >
              <circle
                cx="256"
                cy="256"
                fill="currentColor"
                r="40"
                strokeWidth="0"
                transform="inherit"
                transform-origin="center"
              />
            </svg>
          </g>
          <g
            style={
              Object {
                "opacity": 0.25,
                "transform": "scale(1)",
                "transformOrigin": "center",
                "transition": "opacity 200ms ease-in-out, transform 100ms ease-in-out",
              }
            }
          >
            <svg
              viewBox="0 0 51 48"
            >
              <path
                d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
                fill="currentColor"
                stroke="currentColor"
              />
            </svg>
          </g>
        </svg>
        <svg
          height="3rem"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "cursor": "auto",
            }
          }
          width="3rem"
        >
          <g
            style={
              Object {
                "opacity": 0,
                "transition": "opacity 100ms ease-in-out",
              }
            }
            x={-0.2}
            y={0.9}
          >
            <svg
              viewBox="0 0 512 512"
            >
              <circle
                cx="256"
                cy="256"
                fill="currentColor"
                r="40"
                strokeWidth="0"
                transform="inherit"
                transform-origin="center"
              />
            </svg>
          </g>
          <g
            style={
              Object {
                "opacity": 0.25,
                "transform": "scale(1)",
                "transformOrigin": "center",
                "transition": "opacity 200ms ease-in-out, transform 100ms ease-in-out",
              }
            }
          >
            <svg
              viewBox="0 0 51 48"
            >
              <path
                d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
                fill="currentColor"
                stroke="currentColor"
              />
            </svg>
          </g>
        </svg>
      </div>
    </div>
  </div>
</div>
`;
